### DISCUSSION TITLE
[@types/ldapjs] dn module missing

### DISCUSSION THREAD

**Discussion by agraven** [DefinitelyTyped/DefinitelyTyped#61879] [Issues with a @types package]
The [dn module](https://github.com/ldapjs/node-ldapjs/blob/v2.2.4/lib/index.js#L43) exported by ldapjs is not covered by the type definitions in @types/ldapjs, making the module inaccessible without resorting to `any` coercions or similar

[a comment by an automated account is not shown]

**Comment by pmoleri** [comment]
Hi @agraven , are you willing to create a PR for this? [How can I contribute?](https://github.com/DefinitelyTyped/DefinitelyTyped#how-can-i-contribute)

In order to export the whole dn module, you'll need to declare the classes `DN` and `RDN` which will be some work :)

**Reply by agraven** [reply]
I am willing, though only 90% certain I can dedicate the time to it. I'll let you know when I know.

**Reply by agraven** [reply]
Working on adding this now, am almost done. It took less time than I thought it would :)

**Reply by agraven** [reply]
Filed at #61916
